Framework/Algorithms/src/DiscusMultipleScatteringCorrection.cpp

```diff
@@ -249,9 +258,9 @@ void DiscusMultipleScatteringCorrection::exec() {
   const std::string reportMsg = "Computing corrections";
 
   bool enableParallelFor = true;
-  for (auto &ws : simulationWSs) {
-    enableParallelFor = enableParallelFor && Kernel::threadSafe(*ws);
-  }
+  enableParallelFor = std::all_of(simulationWSs.begin(), simulationWSs.end(),
+                                  [](MatrixWorkspace_sptr ws) { return Kernel::threadSafe(*ws); });
```

Comment:
Could `cbegin()` and `cend()` be used here? Could `ws` also be a ref?

Reply:
Yes I'll change to cbegin\cend. 
Re your other point - are you wondering if the lambda should have `MatrixWorkspace_sptr& ws` or `const MatrixWorkspace_sptr& ws`? Since ws is actually a pointer I'm not sure passing it by reference would add much?

Reply:
It's not a lot, but it would stop there being a copy made of the pointer, which is a small memory save if nothing else. I think. I'll admit I'm not the most clued up on when to use pointers/references/etc. 😅

Reply:
Since this is a smart pointer I think you're right. Taking a copy of the smart pointer probably is more heavy than copying a memory location (which is what you get when passing a raw pointer or doing a pass by reference). I guess some reference counts perhaps gets incremented\decremented with the existing code as well which is further slowness. So will change!